### PR TITLE
Removed backwards compatiblity mode

### PR DESCRIPTION
**Pull request by J3fftw1**
## Description

This PR will remove backwards compatiblity mode.
This will be a start of removing old crap to clean up Slimefun code and make new code more sustainable.

Only 2% (58 servers of the 2301 servers) of servers use backwards compatiblity mode. Where i dont think they know they have it on or have given it a thought. We also recommend in our optimizations guide to turn this feature of.
This should speed up those older servers severaly as well.
I havent benchedmarked this at all

## Proposed changes

Removed unused imports
Deleted backwards compatiblity mode
Deleted test regarded backwards compatiblity mode
Bit of Summer cleaning on my touched files

## Checklist

- [ ] I have fully tested the proposed changes and promise that they will not break everything into chaos.
- [ ] I have also tested the proposed changes in combination with various popular addons and can confirm my changes do not break them.
- [ ] I have made sure that the proposed changes do not break compatibility across the supported Minecraft versions (1.16.* - 1.20.*).
- [ ] I followed the existing code standards and didn't mess up the formatting.
- [ ] I did my best to add documentation to any public classes or methods I added.
- [ ] I have added `Nonnull` and `Nullable` annotations to my methods to indicate their behaviour for null values
- [ ] I added sufficient Unit Tests to cover my code.

### CODE REVIEW

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/tasks/ArmorTask.java [hunk 1]
```diff
@@ -130,8 +130,7 @@ private void handleSlimefunArmor(Player p, ItemStack[] armor, HashedArmorpiece[]
     private void checkForSolarHelmet(@Nonnull Player p) {
         ItemStack helmet = p.getInventory().getHelmet();
 
-        if (Slimefun.getRegistry().isBackwardsCompatible() && !SlimefunUtils.isItemSimilar(helmet, SlimefunItems.SOLAR_HELMET, true, false)) {
-            // Performance saver for slow backwards-compatible versions of Slimefun
+        if (!SlimefunUtils.isItemSimilar(helmet, SlimefunItems.SOLAR_HELMET, true, false)) {
```
Comment (WalshyDev) [inline src/main/java/io/github/thebusybiscuit/slimefun4/implementation/tasks/ArmorTask.java:133]: remove this if block

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/tasks/armor/SolarHelmetTask.java [hunk 2]
```diff
@@ -39,8 +35,7 @@ protected void onPlayerTick(Player p, PlayerProfile profile) {
     private void checkForSolarHelmet(@Nonnull Player p) {
         ItemStack helmet = p.getInventory().getHelmet();
 
-        if (Slimefun.getRegistry().isBackwardsCompatible() && !SlimefunUtils.isItemSimilar(helmet, SlimefunItems.SOLAR_HELMET, true, false)) {
-            // Performance saver for slow backwards-compatible versions of Slimefun
+        if (!SlimefunUtils.isItemSimilar(helmet, SlimefunItems.SOLAR_HELMET, true, false)) {
```
Comment (WalshyDev) [inline src/main/java/io/github/thebusybiscuit/slimefun4/implementation/tasks/armor/SolarHelmetTask.java:38]: same here

File: src/test/java/io/github/thebusybiscuit/slimefun4/implementation/items/TestSlimefunItem.java [hunk 3]
```diff
@@ -105,37 +105,6 @@ void testRecipeType() {
         Assertions.assertThrows(IllegalArgumentException.class, () -> item.setRecipeType(null));
     }
 
-    @ParameterizedTest
-    @DisplayName("Test SlimefunItem#isItem(...)")
-    @ValueSource(booleans = { true, false })
-    void testIsItem(boolean compatibility) {
```
Comment (WalshyDev) [inline src/test/java/io/github/thebusybiscuit/slimefun4/implementation/items/TestSlimefunItem.java:111]: keep this test, just remove this param and usages

**Review by TheBusyBiscuit — APPROVED** [review]
👍🏻 thanks for removing this.
I agree its used rarely enough to justify its removal.

Please do not forget to update the metrics module

**Review by WalshyDev — APPROVED** [review]

**Pull request merged**